templates/pkg/resource/resource.go.tpl

```diff
@@ -88,3 +90,15 @@ func (r *resource) DeepCopy() acktypes.AWSResource {
 	koCopy := r.ko.DeepCopy()
 	return &resource{koCopy}
 }
+
+// IsSynced returns true if the resource is synced.
+func (r *resource) IsSynced() bool {
```

Comment:
Couple things...

a) You should not add a new exported method to Resource without a corresponding change to the AWSResource interface in the runtime repo.

b) I actually do not think that the AWSResource interface is the right interface to modify, because there are some controllers that may need to call alternate AWS APIs to determine whether a resource is synced. And because of that need to potentially call an AWS API, you will need access to the aws-sdk-go SDKAPI object and that is in the AWSResourceManager interface (the one that implements ReadOne, Create, Update, Delete, etc).

Reply:
> b) I actually do not think that the AWSResource interface is the right interface to modify, because there are some controllers that may need to call alternate AWS APIs to determine whether a resource is synced. And because of that need to potentially call an AWS API, you will need access to the aws-sdk-go SDKAPI object and that is in the AWSResourceManager interface (the one that implements ReadOne, Create, Update, Delete, etc).

Fair points, I initially added `IsSynced` to the `AWSResourceDescriptor` interface, but I eventually realized that `resourceDescriptor` doesn't have access to the `Spec/Status` field of an object.
What do you think about the signature of `IsSynced(context.Context, AWSResource) bool` as part of the `AWSResourceManager`

Reply:
> What do you think about the signature of IsSynced(context.Context, AWSResource) bool as part of the AWSResourceManager

++ @A-Hilaly